.github/workflows/build-deb.yaml

```diff
@@ -108,6 +111,101 @@ jobs:
           escaped_json=$(echo "${modified_files}" | jq '.| tostring')
           echo "modified_files=${escaped_json}" >> "${GITHUB_OUTPUT}"
 
+  synchronize-packaging-branches:
+    name: Update packaging branch
+    runs-on: ubuntu-latest
+    needs:
+      - define-versions
+      - build-deb-package
+    permissions:
+      contents: write
+    strategy:
+      fail-fast: false
+      matrix:
+        ubuntu-version: ${{ fromJSON(needs.define-versions.outputs.ubuntu-versions) }}
+    env:
+        PACKAGING_BRANCH: ubuntu-packaging-${{ matrix.ubuntu-version }}
+
+    # Run only on:
+    #  - Push events to main
+    #  - On new tags
+    #  - On github release
+    if: ${{ (github.event_name == 'push' && github.ref == 'refs/heads/main') ||
+       startsWith(github.ref, 'refs/tags/') ||
+       github.event_name == 'release' }}
+
+    steps:
+    - name: Download artifacts
+      uses: actions/download-artifact@v4
+      with:
+        run-id: ${{ needs.build-deb-package.outputs.run-id }}
+        merge-multiple: true
+
+    - name: Install dependencies
+      run: |
+        set -euo pipefail
+
+        sudo apt update -y
+        sudo apt install -y dpkg-dev
+
+    # FIXME: Use dynamic outputs when possible: https://github.com/actions/runner/pull/2477
+    - name: Setup job variables
+      run: |
+        set -exuo pipefail
+
+        json_output='${{ toJSON(needs.build-deb-package.outputs) }}'
+        outputs=(
+          pkg-dsc
+        )
+
+        for var in "${outputs[@]}"; do
+          versioned_var="${var}-${{ matrix.ubuntu-version }}"
+          v=$(echo "${json_output}" | jq -r '."'"${versioned_var}"'"')
+          echo "${var//-/_}=${v}" >> "${GITHUB_ENV}"
+        done
+
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 100
+        path: repo
+
+    - name: Extract the debian sources
+      run: |
+        set -euo pipefail
+
+        dpkg-source -x ${{ env.pkg_dsc }} sources
+
+    - name: Commit packaging sources
+      run: |
+        set -euo pipefail
+
+        # Create or switch to the packaging branch
+        if git -C repo fetch --depth=1 origin "${{ env.PACKAGING_BRANCH }}:${{ env.PACKAGING_BRANCH }}"; then
+          git -C repo checkout "${{ env.PACKAGING_BRANCH }}"
+        else
+          git -C repo checkout -b "${{ env.PACKAGING_BRANCH }}"
+        fi
+
+        # Replace the repository content with the package sources
+        mv repo/.git sources/
+        cd sources
+
+        git config --global user.name "Ubuntu Enterprise Desktop"
+        git config --global user.email "ubuntu-devel-discuss@lists.ubuntu.com"
+
+        git add --all
+        git commit \
+          --allow-empty \
+          -m "Update ubuntu ${{ matrix.ubuntu-version }} package sources" \
+          -m "Use upstream commit ${GITHUB_SHA::7}"
```

Comment:
Limiting the SHA to 7 characters could lead to an ambiguous ID. We could just include the whole SHA here, no?

Reply:
Fine, it's not a big deal as the risk is really low and this is not a place where we may have any trouble, but we've enough space so I'll just go long.